test-harness/README.md

```diff
@@ -10,46 +10,127 @@ In addition, the cobalt test suite allows for better collaboration with embeddin
 
-        ```script
-        ./test-harness/build-base-image.sh -g "<go_version>" -t "<terraform_version>"
-        ```
+- The base image will be tagged as:
 
-        Keep in mind that the terraform version should align with the version from the provider [module](/infra/modules/providers/azure/provider/main.tf#L6)
-    - The base image will be tagged as
+```script
+msftcse/cobalt-test-base:g${GO_VERSION}t${TERRAFORM_VERSION}
+```
 
-        ```script
-        msftcse/cobalt-test-base:g${GO_VERSION}t${TERRAFORM_VERSION}
-        ```
+#### Local Run Script
 
-3. **Local Run** - Run the test runner by calling the below script from the project's root directory.
-    - Test Harness Script
+Run the test runner by calling the below script from the project's root directory. This is one of two options.
 
-        ```script
-        ./test-harness/local-run.sh
-        ```
+```script
+./test-harness/local-run.sh
+```
+
+##### Script Arguments
+
+- `-t` | `--template_name_override`: The template folder to include for the test harness run(i.e. -t "azure-simple-hw"). When set, the git log will be ignored. **Defaults** to the git log.
+- `-b` | `--docker_base_image_name`: The base image to use for the test harness continer. **Defaults** to `msftcse/cobalt-test-base:g${GO_VERSION}t${TF_VERSION}`.
+
+### Option 2: Manual Setup
+
+The benefit with setting up the test harness manually is that runtimes are quicker as we're not rebuilding the test harness image on each run.
+
+The clear downside here is that you'll need to set up all cobalt base software packages and responsible for managing version dependency upgrades over time. Our central base image in docker hub is supported by CSE as well as version dependency upgrades.
+
+The other downside is that you'll need to install this project within your `GOPATH` and pull down all `dep` vendor dependency packages.
+
+#### Prerequisites
+
+- An Azure subscription
+- A [service principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+- An azure storage account for tracking terraform remote backend state. You can use our backend state setup [template](../infra/templates/backend-state-setup/README.md) to provision the storage resources.
+- [git](https://www.atlassian.com/git/tutorials/install-git)
+- Follow [these instructions](https://golang.org/doc/install#download) to download the Go Distribution.
+- Follow these [instructions](https://golang.org/doc/install#testing) to test your golang install.
+- Create your working directory within your `GOPATH`. For example my path for cobalt is:
+
+    ```script
+    $ echo $GOPATH
+    C:\Users\erisch\go
+    $ pwd
+    /c/Users/erisch/go/src/cobalt
+    ```
+
+- Install [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest)
+- Install golang's dep package manager via Git Bash.
```

Comment:
Probably should not assume git bash here, as even on windows people may choose to install this with WSL.

Reply:
the test harness has also been tested on wsl